lib/plugins/create/create.js

```diff
@@ -46,7 +34,79 @@ class Create {
 
   async create() {
     if ('template' in this.options) {
-      return this.createFromTemplate();
+      mainProgress.notice('Setting up a new project', { isMainEvent: true });
+
+      if (!recommendedTemplatesList.includes(this.options.template)) {
+        const errorMessage = [
+          `Template "${this.options.template}" is not supported.`,
+          ` Supported templates are: ${recommendedTemplatesList
+            .map((tmpl) => `"${tmpl}"`)
+            .join(', ')}.`,
+        ].join('');
+        throw new ServerlessError(errorMessage, 'NOT_SUPPORTED_TEMPLATE');
+      }
+
+      try {
+        await downloadTemplateFromExamples({
+          template: this.options.template,
+          name: this.options.name,
+          path: this.options.path,
+          isLegacy: true,
+        });
+      } catch (err) {
+        // Rethrow if the error is about trying to override an existing folder
+        if (
+          err.code === 'TARGET_FOLDER_ALREADY_EXISTS' ||
+          err.code === 'TEMPLATE_FILE_ALREADY_EXISTS'
+        ) {
+          throw err;
+        }
+
+        if (err.code === 'ENOENT') {
+          throw new ServerlessError(
+            'Could not find provided template. Ensure that the template provided with "--template" exists.',
+            'INVALID_TEMPLATE'
+          );
+        }
+
+        if (err.code === 'EACCESS') {
+          const errorMessage = [
+            'Error unable to create a service in this directory. ',
+            'Please check that you have the required permissions to write to the directory',
+          ].join('');
+
+          throw new ServerlessError(errorMessage, 'UNABLE_TO_CREATE_SERVICE');
+        }
+
+        if (err.constructor.name !== 'ServerlessError') throw err;
+
+        throw new ServerlessError(
+          `Could not download template. Ensure that you are using the latest version of Serverless Framework: ${err.message}`,
+          'TEMPLATE_DOWNLOAD_FAILED'
+        );
+      }
+
```

Comment:
Why not place this error handling in `downloadTemplateFromExamples`?

It feels inconsistent to see some errors being prepared there, and some externally

Reply:
good point - in the beginning I added them here as I was trying to reuse original download but then decided against it